libraries/AP_VideoTX/AP_Tramp.cpp

```diff
@@ -0,0 +1,487 @@
+            if (vtx.get_band_and_channel(freq, band, channel)) {
+                vtx.set_band(band);
+                vtx.set_channel(channel);
+            }
+
+            vtx.set_power_mw(power);
+            if (pit_mode) {
+                vtx.set_options(vtx.get_options() | uint8_t(AP_VideoTX::VideoOptions::VTX_PITMODE));
+            } else {
+                vtx.set_options(vtx.get_options() & ~uint8_t(AP_VideoTX::VideoOptions::VTX_PITMODE));
+            }
+
+            // make sure the configured values now reflect reality
+            vtx.set_defaults();
+
+            debug("device config: freq: %u, power: %u, pitmode: %u",
+                unsigned(freq), unsigned(power), unsigned(pit_mode));
+
+
+            return 'v';
+        }
+        break;
+    }
+    case 's': {
+        const uint16_t temp = (int16_t)(response_buffer[6]|(response_buffer[7] << 8));
+        // Check we're not reading the request (indicated by temp zero)
+        if (temp != 0) {
+            // Got response, update device status
+            cur_temp = temp;
+            return 's';
+        }
+        break;
+    }
+    }
+
+    // Likely reading a request, return zero to indicate not accepted
+    return 0;
+}
+
+// Reset receiver state machine
+void AP_Tramp::reset_receiver(void)
+{
+    receive_state = ReceiveState::S_WAIT_LEN;
+    receive_pos = 0;
+}
+
+// returns completed response code or 0
+char AP_Tramp::receive_response()
+{
+    if (port == nullptr) {
+        return 0;
+    }
+
+    // wait for complete packet
+    const uint16_t bytesNeeded = TRAMP_BUF_SIZE - receive_pos;
+    if (port->available() < bytesNeeded) {
+        return 0;
+    }
+
+    for (uint16_t i = 0; i < bytesNeeded; i++) {
+        const uint8_t c = port->read();
+        response_buffer[receive_pos++] = c;
+
+        switch (receive_state) {
+        case ReceiveState::S_WAIT_LEN: {
+            if (c == 0x0F) {
+                // Found header byte, advance to wait for code
+                receive_state = ReceiveState::S_WAIT_CODE;
+            } else {
+                // Unexpected header, reset state machine
+                reset_receiver();
+            }
+            break;
+        }
+        case ReceiveState::S_WAIT_CODE: {
+            if (c == 'r' || c == 'v' || c == 's') {
+                // Code is for response is one we're interested in, advance to data
+                receive_state = ReceiveState::S_DATA;
+            } else {
+                // Unexpected code, reset state machine
+                reset_receiver();
+            }
+            break;
+        }
+        case ReceiveState::S_DATA: {
+            if (receive_pos == TRAMP_BUF_SIZE) {
+                // Buffer is full, calculate checksum
+                const uint8_t cksum = checksum(response_buffer);
+
+                // Reset state machine ready for next response
+                reset_receiver();
+
+                if ((response_buffer[14] == cksum) && (response_buffer[15] == 0)) {
```

Comment:
should be TRAMP_BUF_SIZE-2 and TRAMP_BUF_SIZE-1

Reply:
Fixed